contactaccessrights/tests/phpunit/CRM/Contactaccessrights/Service/ContactRightsTest.php

```diff
@@ -0,0 +1,137 @@
+    $location2 = $this->createOptionValue($this->locationEntityType, 'Location 2');
+
+    // Add Regions / Locations to contact's Rights
+    $regionRights1 = $this->setContactRights($contact, $region1, $this->regionEntityType);
+    $locationRights1 = $this->setContactRights($contact, $location1, $this->locationEntityType);
+    $locationRights2 = $this->setContactRights($contact, $location2, $this->locationEntityType);
+
+    $rights = $this->contactRightsService->getContactRightsByLocations($contact['id']);
+
+    //contact has rights to two locations only.
+    $this->assertCount(2, $rights);
+    $expectedArray = [
+      $locationRights1['id'] => [
+        'id' => $locationRights1['id'],
+        'contact_id' => $contact['id'],
+        'entity_type' => $this->locationEntityType,
+        'entity_id' => $location1['id'],
+        'label' => $location1['label'],
+        'value' => $location1['value']
+      ],
+
+      $locationRights2['id'] => [
+        'id' => $locationRights2['id'],
+        'contact_id' => $contact['id'],
+        'entity_type' => $this->locationEntityType,
+        'entity_id' => $location2['id'],
+        'label' => $location2['label'],
+        'value' => $location2['value']
+      ]
+    ];
+
+    $this->assertEquals($expectedArray, $rights);
+  }
+
+  public function testGetContactRightsByRegions() {
+    $contact =['id' => 5];
+
+    // Create Regions and Locations
+    $region1 = $this->createOptionValue($this->regionEntityType, 'Region 1');
+    $region2 = $this->createOptionValue($this->regionEntityType, 'Region 2');
+    $location1 = $this->createOptionValue($this->locationEntityType, 'Location 1');
+
+    // Add Regions / Locations to contact's Rights
+    $regionRights1 = $this->setContactRights($contact, $region1, $this->regionEntityType);
+    $regionRights2 = $this->setContactRights($contact, $region2, $this->regionEntityType);
+    $locationRights1 = $this->setContactRights($contact, $location1, $this->locationEntityType);
+
+    $rights = $this->contactRightsService->getContactRightsByRegions($contact['id']);
+
+    //contact has rights to two regions only.
+    $this->assertCount(2, $rights);
+    $expectedArray = [
+      $regionRights1['id'] => [
+        'id' => $regionRights1['id'],
+        'contact_id' => $contact['id'],
+        'entity_type' => $this->regionEntityType,
+        'entity_id' => $region1['id'],
+        'label' => $region1['label'],
+        'value' => $region1['value']
+      ],
+
+      $regionRights2['id'] => [
+        'id' => $regionRights2['id'],
+        'contact_id' => $contact['id'],
+        'entity_type' => $this->regionEntityType,
+        'entity_id' => $region2['id'],
+        'label' => $region2['label'],
+        'value' => $region2['value']
+      ]
+    ];
+
+    $this->assertEquals($expectedArray, $rights);
+  }
+
+  private function createOptionValue($optionGroupName, $value) {
```

Comment:
Please don't mix helper methods with tests. Put the helper methods at the end of the file, after all the test methods